index.bs

```diff
@@ -2387,6 +2430,90 @@ Note: Invoking this method from a [=browsing context=] where the [=Web Authentic
+but must be {{ArrayBuffer}} values according to the definition of {{PublicKeyCredentialCreationOptions}}.
+
+<xmp class="idl">
+    partial interface PublicKeyCredential {
+        static PublicKeyCredentialCreationOptions getCreationOptionsFromJSON(PublicKeyCredentialCreationOptionsJSON options);
+    };
+
+    dictionary PublicKeyCredentialCreationOptionsJSON {
+        required PublicKeyCredentialRpEntity                    rp;
+        required PublicKeyCredentialUserEntityJSON              user;
+        required Base64URLString                                challenge;
+        required sequence<PublicKeyCredentialParameters>        pubKeyCredParams;
+        unsigned long                                           timeout;
+        sequence<PublicKeyCredentialDescriptorJSON>             excludeCredentials = [];
+        AuthenticatorSelectionCriteria                          authenticatorSelection;
+        DOMString                                               attestation = "none";
+        AuthenticationExtensionsClientInputsJSON                extensions;
+    };
+
+    dictionary PublicKeyCredentialUserEntityJSON {
+        required Base64URLString        id;
+        required DOMString              name;
+        required DOMString              displayName;
+    };
+
+    dictionary PublicKeyCredentialDescriptorJSON {
+        required Base64URLString        id;
+        required DOMString              type;
+        sequence<DOMString>             transports;
+    };
+
+    dictionary AuthenticationExtensionsClientInputsJSON {
+        DOMString                                       appid;
+        DOMString                                       appidExclude;
+        boolean                                         uvm;
+        boolean                                         credProps;
+        AuthenticationExtensionsLargeBlobInputsJSON     largeBlob;
+    };
+
+    dictionary AuthenticationExtensionsLargeBlobInputsJSON {
+        DOMString                       support;
+        boolean                         read;
+        Base64URLString                 write;
+    };
+</xmp>
+
+</div>
+
+### Deserialize Authentication ceremony options - PublicKeyCredential's `getRequestOptionsFromJSON()` Methods ### {#sctn-getRequestOptionsFromJSON}
+
+<div link-for-hint="WebAuthentication/getRequestOptionsFromJSON">
+
+[=[WRPS]=] use this method to convert JSON-compatible serialized representations of options for {{CredentialsContainer/get()|navigator.credentials.get()}} to {{PublicKeyCredentialRequestOptions}}.
+
+Upon invocation, the [=client=] must take the provided options and return an identically-structured object,
+making sure to decode to {{ArrayBuffer}} those {{DOMString}} properties that are encoded with [=base64url encoding=]
+but must be {{ArrayBuffer}} values according to the definition of {{PublicKeyCredentialRequestOptions}}.
+
```

Comment:
Maybe I missed it somewhere else, but is there any description of how errors should be handled if, for example, the provided JSON is deemed invalid by the client? I expect they would just return via exception, however without it being defined we are open to different browser implementation decisions when an RP developer wants consistency.

The same comment applies to the `getRequestOptionsFromJSON()` method.

Reply:
I added the following to both sections, if we hash it out here I'll make the same changes in the other section:

> If the [=client=] encounters any issues parsing any of the [=JSON type=] representations then it
MUST return an error code equivalent to "{{EncodingError}}" with a description of the incompatible
value and terminate the operation.